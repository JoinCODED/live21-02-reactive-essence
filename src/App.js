import CurrencyInput from "./components/CurrencyInput";
import { AppWrapper } from "./components/styles";

export const USD_TO_EUR_RATE = 0.82;
export const EUR_TO_USD_RATE = 1.22;

const App = () => {
  return (
    <AppWrapper>
      <h2>Quickly convert between USD and EUR ðŸ’°</h2>

      <CurrencyInput icon="ðŸ’µ" />
      <CurrencyInput icon="ðŸ’¶" />
    </AppWrapper>
  );
};

export default App;
