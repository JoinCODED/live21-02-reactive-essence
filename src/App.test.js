/**
 *
 *
 *
 *
 *
 *
 *
 *
 *
 * DO NOT MODIFY THIS FILE
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 */

import { render, fireEvent } from "@testing-library/react";
import App, { USD_TO_EUR_RATE, EUR_TO_USD_RATE } from "./App";

describe("app", () => {
  test("renders correctly", () => {
    const app = render(<App />);

    expect(app.container).toMatchSnapshot();
  });

  test("converts USD to EUR by USD_TO_EUR_RATE", () => {
    const app = render(<App />);

    const usdInput = app.getByPlaceholderText("ðŸ’µ");
    const eurInput = app.getByPlaceholderText("ðŸ’¶");

    let value = 1;
    fireEvent.change(usdInput, { target: { value: value } });
    expect(Number(usdInput.value)).toEqual(value);
    expect(Number(eurInput.value)).toEqual(value * USD_TO_EUR_RATE);

    value = 12;
    fireEvent.change(usdInput, { target: { value: value } });
    expect(Number(usdInput.value)).toEqual(value);
    expect(Number(eurInput.value)).toEqual(value * USD_TO_EUR_RATE);
  });

  test("converts EUR to USD by EUR_TO_USD_RATE", () => {
    const app = render(<App />);

    const usdInput = app.getByPlaceholderText("ðŸ’µ");
    const eurInput = app.getByPlaceholderText("ðŸ’¶");

    let value = 1;
    fireEvent.change(eurInput, { target: { value: value } });
    expect(Number(eurInput.value)).toEqual(value);
    expect(Number(usdInput.value)).toEqual(value * EUR_TO_USD_RATE);

    value = 12;
    fireEvent.change(eurInput, { target: { value: value } });
    expect(Number(eurInput.value)).toEqual(value);
    expect(Number(usdInput.value)).toEqual(value * EUR_TO_USD_RATE);
  });
});
