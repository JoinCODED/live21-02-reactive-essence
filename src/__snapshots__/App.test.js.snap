// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ jexaJZ"
  >
    <h2>
      Quickly convert between USD and EUR ðŸ’°
    </h2>
    <div
      class="sc-gsTCUz kforPu"
    >
      <label
        class="sc-dlfnbm bKRewz"
      >
        ðŸ’µ
      </label>
      <input
        class="sc-hKgILt eUZztZ"
        placeholder="ðŸ’µ"
        type="number"
        value="1"
      />
    </div>
    <div
      class="sc-gsTCUz kforPu"
    >
      <label
        class="sc-dlfnbm bKRewz"
      >
        ðŸ’¶
      </label>
      <input
        class="sc-hKgILt eUZztZ"
        placeholder="ðŸ’¶"
        type="number"
        value="0.82"
      />
    </div>
  </div>
</div>
`;
